pipeline/ingest/ephys.py

```diff
@@ -250,116 +210,263 @@ def _load(self, data, probe, npx_meta, rigpath):
                                                     'shank_col': shank_col + 1,
                                                     'shank_row': shank_row + 1}).fetch1('KEY')
 
-        # insert Unit
-        log.info('.. ephys.Unit')
-
-        with InsertBuffer(ephys.Unit, 10, skip_duplicates=True,
-                          allow_direct_insert=True) as ib:
-
-            for i, u in enumerate(set(units)):
-                if method in ['jrclust_v3', 'jrclust_v4']:
-                    wf_chn_idx = 0
-                elif method in ['kilosort2']:
-                    wf_chn_idx = np.where(data['ks_channel_map'] == vmax_unit_site[i])[0][0]
-
-                ib.insert1({**skey, **insertion_key,
-                            **site2electrode_map[vmax_unit_site[i]],
-                            'clustering_method': method,
-                            'unit': u,
-                            'unit_uid': u,
-                            'unit_quality': unit_notes[i],
-                            'unit_posx': unit_xpos[i],
-                            'unit_posy': unit_ypos[i],
-                            'unit_amp': unit_amp[i],
-                            'unit_snr': unit_snr[i],
-                            'spike_times': unit_spikes[i],
-                            'waveform': unit_wav[i][wf_chn_idx]})
-
-                if ib.flush():
-                    log.debug('.... {}'.format(u))
-
-        # insert Unit.UnitTrial
-        log.info('.. ephys.Unit.UnitTrial')
-        dj.conn().ping()
-        with InsertBuffer(ephys.Unit.UnitTrial, 10000, skip_duplicates=True,
-                          allow_direct_insert=True) as ib:
+        if ininto_archive:
```

Comment:
since these 2 blocks are so large, it might make sense to break out archive/non archive insert to separate functions or maybe define as inner functions or maybe collapse directly e.g:
```
if into_archive:
    unit_table = ephys.ArchivedClustering.Unit
 else:
    unit_table = ephys.Unit
```
etc. 
For the ones that only apply to archive, perhaps set them to None and test on insert - like:
```
if archived_clustering_table:
    archived_clustering_table.insert1(...)
```
maybe just the separate functions is best?